core/src/main/java/org/eclipse/hono/util/RegistrationConstants.java

```diff
@@ -158,17 +160,20 @@ public static JsonObject getRegistrationJson(final String action, final String t
     }
 
     public static JsonObject getRegistrationJson(final String action, final String tenantId, final String deviceId, final JsonObject payload) {
-        return getRegistrationJson(action, tenantId, deviceId, null, payload);
+        return getRegistrationJson(action, tenantId, deviceId, null, null, payload);
     }
 
-    public static JsonObject getRegistrationJson(final String action, final String tenantId, final String deviceId, final String key, final JsonObject payload) {
+    public static JsonObject getRegistrationJson(final String action, final String tenantId, final String deviceId, final String key, final String value, final JsonObject payload) {
         final JsonObject msg = new JsonObject();
         msg.put(APP_PROPERTY_ACTION, action);
         msg.put(APP_PROPERTY_DEVICE_ID, deviceId);
         msg.put(APP_PROPERTY_TENANT_ID, tenantId);
         if (key != null) {
             msg.put(APP_PROPERTY_KEY, key);
         }
+        if (value != null) {
```

Comment:
It depends if we want to search only for key. But to keep it simple at the beginning I agree to allow only key together with value. I just found a possible other problem with this - I added a device with a complex structure behind a key. Not only a simple type value. In this case I would expect, that it is not possible to search for this or that you need to give the whole "value" as a JSON structure. I found, that none of then is possible, because as soon there is a complex structure there will be a casting problem in the find implementation (for every search). I suggest to fix this too with this pull request, but the question remains, what the expected behavior is. I suggest to ignore complex values at find (at the moment). What do you think?